### PR TITLE
ICDS Dashboard: add maintenance alerts to mobile dashboard

### PR DESCRIPTION
**Pull request by czue**
https://trello.com/c/XKhFoWFv/195-maintenance-alerts

image

Easy, review how you like

### CODE REVIEW

File: custom/icds_reports/templates/icds_reports/mobile/dashboard/mobile_dashboard.html [hunk 1]
```diff
@@ -102,5 +103,20 @@
     minDuration: 700,
   });
   angular.bootstrap(document.getElementById('app'), [appName]);
+  $(function () {
+    // maintenance alert hide/show logic - lifted from hqwebapp/js/main.js
```
Comment (czue) [inline custom/icds_reports/templates/icds_reports/mobile/dashboard/mobile_dashboard.html]: this is more code than I'd normally like to copy / paste though wasn't sure exactly how to extract it in the easiest/cleanest way. open to suggestions if you have any.

**Review by orangejenny — COMMENTED** [review]
Looks good, not approving yet because of the question on `filters.less`

File: custom/icds_reports/static/icds_reports/mobile/css/filters.less [hunk 2]
```diff
@@ -99,7 +99,8 @@
   top: -5px;
   text-align: center;
 }
-.alert {
+
+.filter-alert {
```
[the remark this answers is not in the document]
Comment (czue) [reply custom/icds_reports/static/icds_reports/mobile/css/filters.less]: Good question. Yeah, this was added [a week ago](https://github.com/dimagi/commcare-hq/pull/26351) and is not used anywhere else.

File: corehq/apps/hqwebapp/templatetags/hq_shared_tags.py [hunk 3]
```diff
@@ -447,7 +447,7 @@ def maintenance_alert(request, dismissable=True):
             ' hide' if dismissable else '',
             alert.id,
             mark_safe('''
-                <a href="#" class="close" data-dismiss="alert" aria-label="close">&times;</a>
+                <a class="close" data-dismiss="alert" aria-label="close">&times;</a>
```
[the remark this answers is not in the document]
Comment (czue) [reply corehq/apps/hqwebapp/templatetags/hq_shared_tags.py]: Good call. https://github.com/dimagi/commcare-hq/pull/26403/commits/e5e0b7836430be3dcd32948b597259e9c3afef7e

**Review by orangejenny — APPROVED** [review]

File: custom/icds_reports/templates/icds_reports/mobile/dashboard/mobile_dashboard.html (hunk 1 above)
Comment (czue) [reply custom/icds_reports/templates/icds_reports/mobile/dashboard/mobile_dashboard.html]: Roger, thanks. I added a [backlog item](https://trello.com/c/aJr4COdX/301-fix-duplication-of-maintenance-alerts-code) for this that I will hopefully get to in not too long.

File: custom/icds_reports/static/icds_reports/mobile/css/app.less [hunk 4]
```diff
@@ -324,3 +323,7 @@
   margin-bottom: inherit;
   border: 0;
 }
+
+.alert-maintenance .close {
```
[the remark this answers is not in the document]
Comment (czue) [reply custom/icds_reports/static/icds_reports/mobile/css/app.less]: HQ already adds the `alert` class actually (https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/hqwebapp/templatetags/hq_shared_tags.py#L446) but you were right that the padding was redundant. Fixed in e8164e9e2eeade28b239b7b41c678a92089c0667. everything else is actually overriding the behavior of that class that doesn't work on mobile dashboard.

[the remark this answers is not in the document]
Comment (orangejenny) [reply custom/icds_reports/static/icds_reports/mobile/css/app.less]: Cool - re-reading, my comment was a misunderstanding, I had confused `alert-maintenance` and `filter-alert`.

**Pull request merged**